sdk/dotnet/Pulumi/Deployment/Deployment.Runner.cs

```diff
@@ -117,7 +117,11 @@ async void HandleCompletion(Task task)
                             await task.ConfigureAwait(false);
 
                             // Log the descriptions of completed tasks.
-                            var descriptions = _inFlightTasks[task];
+                            List<string> descriptions;
+                            lock (_inFlightTasks)
```

Comment:
Not suggesting a change for now, but just wondering why isn't `_inFlightTasks` typed as a `ConcurrentDictionary`? Then we wouldn't need the synchronization code around its use in this class.

Reply:
Yeah I had this thought more than once but haven't tried it - not sure if it supports all the same operations with same semantics, but I guess it would be a good change.